src/errors.ts

```diff
@@ -93,79 +82,70 @@ export class CustomError extends Error {
-    const resp: Response = responseError.response;
-    const errorBody: string = await resp.clone().text();
-    logErrorMessage = `Error response: ${message}`;
+    const error = e as AnyResponseError;
+    const errorBody = await error.response.clone().text();
+    errorMessage = `[${messagePrefix}] error response:`;
     errorContext = {
-      responseStatus: resp.status,
-      responseStatusText: resp.statusText,
+      responseStatus: error.response.status,
+      responseStatusText: error.response.statusText,
       responseBody: errorBody.slice(0, 5000), // limit to 5000 characters
-      responseErrorType: responseError.name,
+      responseErrorType: error.name,
     };
-    responseStatusCode = resp.status;
-    wrappedError = new CustomError(
-      responseError.name,
-      `Error ${resp.status} "${resp.statusText}" @ ${resp.url}: ${message}`,
-      responseError,
-    );
+    responseStatusCode = error.response.status;
   } else {
-    // non-ResponseError error
-    logErrorMessage = `Error: ${message} --> ${e}`;
-    errorContext = {
-      errorType: e.name,
-      errorMessage: e.message,
-      errorStack: e.stack,
-    };
-    wrappedError = new CustomError(e.name, `${message}: ${e.message}`, e);
-  }
-
-  // also handle any nested errors from either the ResponseError or (more likely) other Errors
-  if (hasErrorCause(e)) {
-    const errorChain = getNestedErrorChain(e.cause);
-    if (errorChain.length) {
+    // something we caught that wasn't actually a ResponseError type but was passed in here anyway
+    errorMessage = `[${messagePrefix}] error: ${e}`;
+    if (e instanceof Error) {
+      // top-level error
       errorContext = {
-        ...errorContext,
-        errors: JSON.stringify(errorChain, null, 2),
+        errorType: e.name,
+        errorMessage: e.message,
+        errorStack: e.stack,
       };
+      // also handle any nested errors starting from the `cause` property
+      if (hasErrorCause(e)) {
+        const errorChain = getNestedErrorChain(e.cause);
+        if (errorChain.length) {
+          errorContext = {
+            ...errorContext,
+            errors: JSON.stringify(errorChain, null, 2),
+          };
+        }
+      }
     }
   }
 
-  logger.error(logErrorMessage, { ...errorContext, ...extra });
-
+  logger.error(errorMessage, { ...errorContext, ...sentryContext });
   // TODO: follow up to reuse EventHint type for capturing tags and other more fine-grained data
-  if (sendTelemetry) {
-    sentryCaptureException(wrappedError, {
+  const hasMeaningfulSentryContext =
+    [
+      sentryContext.extra,
+      sentryContext.contexts,
+      sentryContext.tags,
+      sentryContext.propagationContext,
+    ].some((context) => context && Object.keys(context).length > 0) ||
+    sentryContext.user?.id !== undefined ||
+    (sentryContext.fingerprint && sentryContext.fingerprint.length > 0);
+
+  if (hasMeaningfulSentryContext) {
+    sentryCaptureException(e, {
```

Comment:
For this section, I think we can just check if `sentryContext` has any keys and then try to merge it into the value of `captureContext` with some safe defaults, like this:
```suggestion
  if (Object.keys(sentryContext).length) {
    sentryCaptureException(e, {
      captureContext: {
        ...sentryContext,
        contexts: {
          ...(sentryContext.contexts ?? {}),
          response: { status_code: responseStatusCode },
        },
        extra: {
          ...(sentryContext.extra ?? {}),
          ...errorContext,
        },
      },
    });
  }
```
So if the caller passes something else in the sentry context like `tags`, it will get automatically included, but we'll merge some of the `logError` provided context with whatever the caller passed for `extra`/`contexts`